TheJumpCloud/support
# Fix hyphen user generation and EmailSAN certificate replacement

## Issues
* [CUT-4504](https://jumpcloud.atlassian.net/browse/CUT-4504) - Users with a hyphen "-" do not get a radius user certificate
* [CUT-4505](https://jumpcloud.atlassian.net/browse/CUT-4505) - EmailSAN Certificates are not replaced on Windows Device

## What does this solve?

- Support for users who have a hyphen "-" character in their username
- Resolves an issue for the replacement of EmailSAN certificates where previously generated certificates were not removed

## Is there anything particularly tricky?

## How should this be tested?

The bugfix for users with a hyphen in their username should be covered by automated tests, review updated test files.

To test the EmailSAN fix:

- On `config.ps1` update the cert type to be EmailSAN: `$Global:JCR_CERT_TYPE = "EmailSAN"` on line 36
- Generate a set of certificates for users, ensure at least one of those users is associated to a windows Virtual Machine.
- Distribute the certificate to the user associated to a windows virtual machine
- When the command runs on the device the user should get a certificate with Subject Alternative Name containing that user's email address: 
![Screenshot 2024-12-10 at 11 35 32 AM](https://github.com/user-attachments/assets/17c2536c-6cd2-4ec4-a72a-645ed595c9b1)
- re-generate a new certificate for that user
- re-distribute that new certificate for that user
- When the command runs the old certificate should be removed from the device and the new certificate should be left in it's place:
![Screenshot 2024-12-10 at 11 37 25 AM](https://github.com/user-attachments/assets/adb3aef0-cedd-4e88-abd5-21c066117531)
Note the thumbprints differ between the screenshots, the cert was replaced and only one single user cert remains in the user's cert store
![Screenshot 2024-12-10 at 11 38 02 AM](https://github.com/user-attachments/assets/a3d1f323-e817-4f9b-b9df-9495eb2aa475)


## Screenshots


[CUT-4504]: https://jumpcloud.atlassian.net/browse/CUT-4504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CUT-4505]: https://jumpcloud.atlassian.net/browse/CUT-4505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
